src/util/postgres_client.js

```diff
@@ -77,8 +76,14 @@ function encode_json(schema, val) {
     if (!val || !schema) {
         return val;
     }
-
     if (schema.binary === true) {
+        if (val.$in) {
```

Comment:
Seems like a patch for dedup_keys.
We should develop something more robust in the future for other use cases as well.
This is relevant for every type and not only for binary.

Reply:
moved it out to a function that handles specific ops that we currently need.